src/js/components/user/user.js

```diff
@@ -53,6 +60,24 @@ class User extends MaharaBaseComponent {
                   <button onClick={this.loginButton} className="big">{this.gettext('wizard_login_button') }</button>
                 </section>;
         } else {
+          let langOptions = Object.keys(window.mahara.i18n.strings);
+          let journalOptions = this.props.server.sync.blogs.map(blog =>  {
+                    return {
+                      'id': blog.id,
+                      'text': blog.title
+                    };
+                  });
+          let folderOptions = this.props.server.sync.folders.map(folder =>  {
+                    return {
+                      'id': folder.title,
+                      'text': folder.title
+                    };
+                  });
+          let defaultFolder = this.props.server.defaultFolderName ? folderOptions.find(folder => folder.id === this.props.server.defaultFolderName) : { 'id': '', 'title': ''};
```

Comment:
do these need to cater the the possibility that the map of this.props.server.sync.blogs doesn't have the values you're looking for?